File: 167.two-sum-ii-input-array-is-sorted.py
#
# @lc app=leetcode id=167 lang=python
#
# [167] Two Sum II - Input array is sorted
#

# @lc code=start
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        start = 0
        end = len(numbers) - 1
        while start != end:
            cur = numbers[start] + numbers[end]
            if cur == target:
                return [start+1, end+1]
            elif cur < target:
                start += 1
            else:
                end -= 1
        return None
# @lc code=end

File: 167.two-sum-ii-input-array-is-sorted.cpp
/*
 * @lc app=leetcode id=167 lang=cpp
 *
 * [167] Two Sum II - Input array is sorted
 */
#include <vector>

using namespace std;

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0;
        int r = numbers.size() - 1;
        while(l<r) {
            if (numbers[l] + numbers[r] == target) {
                vector<int> res{l + 1, r + 1};
                return res;
            }
            else if (numbers[l] + numbers[r] > target){
                r--;
            }
            else {
                l++;
            }
        }
        return {};
    }
};
// @lc code=end

